File: example-week/coinchange/submissions/accepted/aditya.py
x = int(input())
MOD = int(1e9 + 7)

coins = [1, 5, 10, 25]

dp = [0 for _ in range(x + 1)]
dp[0] = 1

for i in range(1, x + 1):
    for c in coins:
        if i - c >= 0:
            dp[i] = (dp[i] + dp[i - c]) % MOD
            
print(dp[x])

File: example-week/coinchange/submissions/accepted/aditya.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;

const int MOD = 1e9 + 7;

int main() {
    int x; cin >> x;
    vector<long long> dp(x + 1);
    dp[0] = 1;
    vector<int> coins = {1, 5, 10, 25};
    
    for(int i = 1; i <= x; i++) {
        for(int& c : coins) {
            if(i - c >= 0) {
                dp[i] = (dp[i] + dp[i - c]) % MOD;
            }
        }
    }
    
    cout << dp[x] << endl;
    
    return 0;
}
